File: Amakoye/power_of_two.py
def isPowerOfTwo(n):
    if n == 1:
        return True

    if n == 0 or n % 2 != 0:
        return False
    return isPowerOfTwo(n // 2)


print(isPowerOfTwo(6))
print(isPowerOfTwo(8))
print(isPowerOfTwo(112))
print(isPowerOfTwo(16))

File: joshgog/PowerOfTwo.java
public class PowerOfTwo {
    public static boolean isPowerOfTwo(int n) {
        if (n <= 0) {
            return false;
        }
        return (n & (n - 1)) == 0;
    }

    public static void main(String[] args) {
        int number1 = 8;
        int number2 = 5;
        
        System.out.println(isPowerOfTwo(number1)); //  true
        System.out.println(isPowerOfTwo(number2)); //  false
    }
}
